src/Foundation/NSAttributedStringDocumentAttributes.cs

```diff
@@ -211,15 +212,40 @@ public partial class NSAttributedStringDocumentAttributes : DictionaryContainer
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("maccatalyst")]
 		[UnsupportedOSPlatform ("tvos")]
+		[ObsoletedOSPlatform ("ios17.0", message: "Starting with ios17.0 will always return null.")]
+		[ObsoletedOSPlatform ("maccatalyst17.0", message: "Starting with maccatalyst17.0 will always return null.")]
+		[ObsoletedOSPlatform ("macos14.0", message: "Starting with macos14.0 will always return null.")]
 #else
+		[Deprecated (PlatformName.iOS, 17, 0, message: "Will always return null.")]
+		[Deprecated (PlatformName.MacCatalyst, 17, 0, message: "Will always return null.")]
+		[Deprecated (PlatformName.MacOSX, 14, 0, message: "Will always return null.")]
 		[iOS (13, 0)]
 #endif
 		public NSUrl? ReadAccessUrl {
 			get {
-				return GetNativeValue<NSUrl> (NSAttributedStringDocumentReadingOptionKey.NSReadAccessUrlDocumentOption);
+				return null;
+			}
+			set {
+				// do nothing
+			}
```

Comment:
This seems wrong: NSReadAccessUrlDocumentOption should work on older OSes, so there's no reason to remove the code.